newsroom/agenda/service.py

```diff
@@ -0,0 +1,123 @@
+from datetime import datetime
+from eve.utils import ParsedRequest
+
+from newsroom.auth.utils import get_user_from_request, get_company_from_request
+from newsroom.agenda.agenda import (
+    is_events_only_access,
+    _agenda_query,
+    nested_query,
+    planning_items_query_string,
+    aggregations,
+    _remove_fields,
+    PLANNING_ITEMS_FIELDS,
+)
+from newsroom.agenda.model import FeaturedResourceModel
+from newsroom.core.resources.service import NewshubAsyncResourceService
+from newsroom.utils import get_local_date
+from newsroom.template_filters import is_admin
+
+from superdesk import get_resource_service
+from superdesk.core import json
+from superdesk.flask import abort
+from superdesk.utc import local_to_utc
+from superdesk.utils import ListCursor
+
+
+class FeaturedService(NewshubAsyncResourceService[FeaturedResourceModel]):
+    resource_name = "agenda_featured"
+
+    async def on_create(self, docs):
+        """
+        Add UTC from/to datetimes on save.
+
+        Problem is 31.8. in Sydney is from 30.8. 14:00 UTC to 31.8. 13:59 UTC.
+        And because we query later using UTC, we store those UTC datetimes as
+        display_from and display_to.
+        """
+        for item in docs:
+            date = datetime.strptime(item._id, "%Y%m%d")
+            item.display_from = local_to_utc(item.tz, date.replace(hour=0, minute=0, second=0))
+            item.display_to = local_to_utc(item.tz, date.replace(hour=23, minute=59, second=59))
+        await super().on_create(docs)
+
+    async def find_one_for_date(self, for_date: datetime) -> FeaturedResourceModel | None:
+        return await self.find_one(req=None, display_from={"$lte": for_date}, display_to={"$gte": for_date})
+
+    async def get_featured_stories(self, req, lookup):
+        for_date = datetime.strptime(req.args.get("date_from"), "%d/%m/%Y %H:%M")
+        offset = int(req.args.get("timezone_offset", "0"))
+        local_date = get_local_date(
+            for_date.strftime("%Y-%m-%d"),
+            datetime.strftime(for_date, "%H:%M:%S"),
+            offset,
+        )
+        featured_doc = await self.find_one_for_date(local_date)
+        return await self.featured(req, lookup, featured_doc)
+
+    async def featured(self, req, lookup, featured):
+        """Return featured items.
+
+        :param ParsedRequest req: The parsed in request instance from the endpoint
+        :param dict lookup: The parsed in lookup dictionary from the endpoint
+        :param dict featured: list featured items
+        """
+        user = get_user_from_request(None)
+        company = get_company_from_request(None)
+        if is_events_only_access(user.to_dict(), company.to_dict()):
+            abort(403)
+
+        if not featured or not featured.get("items"):
+            return ListCursor([])
+
+        query = _agenda_query()
+        get_resource_service("section_filters").apply_section_filter(query, self.section)
```

Comment:
Section filters are now available via the new Async service. You can now import it's service and directly use it here instead (the get_resource_service is the old way, and returns the deprecated sync resource)